working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1314,12 +1402,15 @@ The context type schema for a pattern `p` is:
     var [[int x]!] = [[]]; // Infers List<List<int>?> for the list literal.
     ```
 
-*   **Literal** or **constant**: The context type schema is the static type of
-    the pattern's constant value expression.
+*   **Constant**: The context type schema is the static type of the pattern's
+    value.
 
 *   **Variable**:
 
-    1.  If `p` has no type annotation, the context type schema is `?`.
+    1.  In an assignment context, the context type schema is the static type of
```

Comment:
What if the variable is promoted?  Does `static type` mean the promoted type or the unpromoted type of the variable?

I think it should mean the unpromoted type, so that demoting assignments are possible (as they are for ordinary non-pattern assignments).

Reply:
Haha, good question! I'll defer to your expertise. Done. :)

Reply:
Do what we'd do on a plain assignment, `x = e`, for the same variable.